plugins/woocommerce/client/blocks/assets/js/base/context/hooks/cart/test/use-store-cart-coupons.tsx

```diff
@@ -0,0 +1,414 @@
+			fetchMock.mockResponseOnce(
+				JSON.stringify( {
+					responses: [ { status: 200, body: {} } ],
+				} )
+			);
+
+			const { result } = renderHook( () =>
+				useStoreCartCoupons( 'wc/checkout' )
+			);
+
+			await act( async () => {
+				await result.current.removeCoupon( 'REMOVE_TEST' );
+			} );
+
+			// Verify the API call was made correctly for removal
+			expect( fetchMock ).toHaveBeenCalledTimes( 1 );
+
+			const requestBody = fetchMock.mock.calls[ 0 ][ 1 ].body;
+			const parsedBody = JSON.parse( requestBody );
+
+			expect( parsedBody.requests[ 0 ].path ).toBe(
+				'/wc/store/v1/cart/remove-coupon'
+			);
+			expect( parsedBody.requests[ 0 ].method ).toBe( 'POST' );
+			expect( parsedBody.requests[ 0 ].data.code ).toBe( 'REMOVE_TEST' );
+		} );
+	} );
+
+	describe( 'Context-specific behavior', () => {
+		it( 'works correctly with wc/checkout context', async () => {
+			fetchMock.mockResponseOnce(
+				JSON.stringify( {
+					responses: [ { status: 200, body: {} } ],
+				} )
+			);
+
+			// Test with checkout context specifically
+			const { result } = renderHook( () =>
+				useStoreCartCoupons( 'wc/checkout' )
+			);
+
+			await act( async () => {
+				await result.current.applyCoupon( 'CHECKOUT_CONTEXT' );
+			} );
+
+			// Verify API call is made regardless of context
+			expect( fetchMock ).toHaveBeenCalledTimes( 1 );
+
+			const requestBody = fetchMock.mock.calls[ 0 ][ 1 ].body;
+			const parsedBody = JSON.parse( requestBody );
+			expect( parsedBody.requests[ 0 ].data.code ).toBe(
+				'CHECKOUT_CONTEXT'
+			);
+		} );
+
+		it( 'works correctly with different contexts', async () => {
+			fetchMock.mockResponses(
+				[
+					JSON.stringify( {
+						responses: [ { status: 200, body: {} } ],
+					} ),
+					{ status: 200 },
+				],
+				[
+					JSON.stringify( {
+						responses: [ { status: 200, body: {} } ],
+					} ),
+					{ status: 200 },
+				]
+			);
+
+			// Test checkout context
+			const { result: checkoutResult } = renderHook( () =>
+				useStoreCartCoupons( 'wc/checkout' )
+			);
+
+			// Test cart context
+			const { result: cartResult } = renderHook( () =>
+				useStoreCartCoupons( 'wc/cart' )
+			);
+
+			await act( async () => {
+				await checkoutResult.current.applyCoupon( 'CHECKOUT_COUPON' );
+			} );
+
+			await act( async () => {
+				await cartResult.current.applyCoupon( 'CART_COUPON' );
+			} );
+
+			// Both should make the same API calls
+			expect( fetchMock ).toHaveBeenCalledTimes( 2 );
+
+			// Verify both calls have the same API structure
+			const firstCall = JSON.parse( fetchMock.mock.calls[ 0 ][ 1 ].body );
+			const secondCall = JSON.parse(
+				fetchMock.mock.calls[ 1 ][ 1 ].body
+			);
+
+			expect( firstCall.requests[ 0 ].path ).toBe(
+				'/wc/store/v1/cart/apply-coupon'
+			);
+			expect( secondCall.requests[ 0 ].path ).toBe(
+				'/wc/store/v1/cart/apply-coupon'
+			);
+		} );
+	} );
+
+	describe( 'Edge cases', () => {
+		it( 'handles special characters in coupon codes', async () => {
```

Comment:
FYI Mike also added some good tests for special characters here https://github.com/woocommerce/woocommerce/pull/58918/files#diff-48cc2b59a2fa113d4aad9206ba43c684f72624f4c98a20bc19bd3317469b55a2R393